python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -2256,6 +2260,68 @@ def with_partitions_def(
     def __repr__(self) -> str:
         return f"TimeWindowPartitionsSubset({self.get_partition_key_ranges(self.partitions_def)})"
 
+    def __sub__(self, other: "PartitionsSubset") -> "PartitionsSubset":
+        if self is other:
+            return self.empty_subset(self.partitions_def)
+
+        empty_subset = self.empty_subset(self.partitions_def)
+
+        if other is empty_subset:
+            return self
+
+        if not isinstance(other, TimeWindowPartitionsSubset):
+            return empty_subset.with_partition_keys(
+                set(self.get_partition_keys()).difference(set(other.get_partition_keys()))
+            )
+
+        # can we assume these are already sorted chronologically
```

Comment:
we likely can, but it's O(n) to sort an already-sorted list in python anyway, so I prefer this solution as it makes it clear what the invariant is